code/script.js

```diff
@@ -255,12 +261,75 @@ const checkQuestion = () => {
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  // if (category === 'hair' || category === 'eyes') {
+  //   charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
 
-  } else if (category === 'accessories' || category === 'other') {
+  // } else if (category === 'accessories' || category === 'other') {
+  //   charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
 
+  // }
+  // if (category === 'hair' || category === 'eyes') {
+  //   charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
+  // } else if (category === 'accessories') {
+  //   if (value === 'glasses') {
+  //     charactersInPlay = charactersInPlay.filter((person) => person[category].includes('glasses'));
+  //   } else if (value === 'hat') {
+  //     charactersInPlay = charactersInPlay.filter((person) => person[category].includes('hat'));
+  //   }
+  // } else if (category === 'other') {
+  //   if (value === 'smoker') {
+  //     charactersInPlay = charactersInPlay.filter((person) => person[category].includes('smoker'));
+  //   } else {
+  //     // Handle other "other" category values here if needed
+  //   }
+  // }
+
```

Comment:
The logic  in this code currently is the following. 1. Selects "Yellow hair" in filter list -> Board keeps the "Yellow hair" 2. Selects "Glasses" --> Board keeps glasses 3. Secret person may be filtered out and not among the choices.